src/js/tabs/pid_tuning.js

```diff
@@ -1313,14 +1370,29 @@ TABS.pid_tuning.initialize = function (callback) {
             updatePidDisplay();
         });
 
-        $('#resetProfile').on('click', function(){
+        $('#resetProfile').on('click', function() {
             self.updating = true;
-            MSP.promise(MSPCodes.MSP_SET_RESET_CURR_PID).then(function () {
-                self.refresh(function () {
+
+            function refresh () {
+                self.refresh(() => {
                     self.updating = false;
 
                     GUI.log(i18n.getMessage('pidTuningProfileReset'));
                 });
+            }
+
+            MSP.promise(MSPCodes.MSP_SET_RESET_CURR_PID).then(() => {
+                if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44)) {
+                    MSP.promise(MSPCodes.MSP_EEPROM_WRITE).then(() => {
```

Comment:
What is the reason that we are executing an immediate save operation after this?

Reply:
Now changing a slider go to another tab and come back to pidtuning it's as from the start (unsaved changes are lost) .
Saving values here because as long settings are not saved configurator thinks it is until the FC reboots (unsaved changes are retained until reboot and then lost - which could lead to surprises in the field).

Doing the refresh resets variables holding user changes to reset the state without saving. So perhaps I could store and retrieve those variables instead.

Reply:
`MSP_SET_RESET_CURR_PID` is not resetting the Gyro Filter.

Reply:
The problems that I can see with this are:
- it is unexpected to the user that we save settings outside of clicking 'Save';
- the proposed behaviour is inconsistent for different firmware versions;
- this is even more confusing if a TX is connected at the same time and the user is switching profiles through the TX, triggering saves in configurator.

So I think a better way would be to revert to the old behaviour of just discarding the changes (now probably requiring a restoreInitialSettings` on PID profile changes).

Also, the gyro filter is not part of the PID profile, so there is no reason it should be linked in any way to the PID profile functionality.

Reply:
Found a way ;-)
Have removed the tooltip for the button as it's no longer needed. Now it behaves as before.
`initialSettings` were overwritten on tab `reload,` but now have added another flag to workaround.